invokeai/app/invocations/baseinvocation.py

```diff
@@ -615,7 +615,15 @@ def wrapper(cls: Type[GenericBaseInvocationOutput]) -> Type[GenericBaseInvocatio
             config=cls.__config__,
         )
         cls.__fields__.update({"type": output_type_field})
-        cls.__annotations__.update({"type": output_type_annotation})
+
+        # to support 3.9, 3.10 and 3.11, as described in https://docs.python.org/3/howto/annotations.html
+        annotations = (
+            cls.__dict__.get("__annotations__", None)
+            if isinstance(cls, type)
+            else getattr(cls, "__annotations__", None)
+        )
+        if annotations:
+            annotations.update({"type": output_type_annotation})
```

Comment:
This can be simplified; we in a class so `isinstance(cls, type)` is redundant - only the first expression will be evaluated.
```suggestion

        # to support 3.9, 3.10 and 3.11, as described in https://docs.python.org/3/howto/annotations.html
        annotations = cls.__dict__.get("__annotations__", None)
        if annotations:
            annotations.update({"type": output_type_annotation})
```